File: 2nd_Project/_zmq.cpp
#include <zmq.hpp>
#include <iostream>

int main() {
    zmq::context_t context(1);
    zmq::socket_t socket(context, ZMQ_REQ); // 요청-응답 패턴 사용
    socket.connect("tcp://localhost:5555"); // 서버 연결

    std::string message = "Hello from C++";
    zmq::message_t request(message.size());
    memcpy(request.data(), message.c_str(), message.size());

    socket.send(request); // 메시지 전송

    zmq::message_t reply;
    socket.recv(&reply); // 응답 수신
    std::cout << "서버 응답: " << std::string(static_cast<char*>(reply.data()), reply.size()) << std::endl;

    return 0;
}

File: 2nd_Project/_zmq.py
import zmq

# ZeroMQ 컨텍스트 생성
context = zmq.Context()
socket = context.socket(zmq.REP)  # 요청-응답 패턴 사용
socket.bind("tcp://*:5555")  # 포트 5555에서 대기

print("서버 시작...")

while True:
    message = socket.recv_string()  # 클라이언트로부터 메시지 수신
    print(f"받은 메시지: {message}")

    response = f"서버 응답: {message.upper()}"
    socket.send_string(response)  # 응답 전송